.github/workflows/validate-pr.yml

```diff
@@ -49,14 +49,20 @@ jobs:
     runs-on: ubuntu-20.04
     env:
       PR_TITLE: ${{ github.event.pull_request.title }}
+      TARGET_BRANCH: ${{ github.event.pull_request.base.ref }}
     steps:
       # Backports or PR that target a release branch directly should mention the target branch in the title, for example:
       # [X.Y backport] Some change that needs backporting to X.Y
       # [X.Y] Change directly targeting the X.Y branch
       - name: Get branch from PR title
         id: title_branch
-        run: echo "$PR_TITLE" | sed -n 's/^\[\([0-9]*\.[0-9]*\)[^]]*\].*/branch=\1/p' >> $GITHUB_OUTPUT
+        run: |
+          # get the intended major version prefix ("[27.1 backport]" -> "27.") from the PR title.
+          echo "$PR_TITLE" | sed -n 's/^\[\([0-9]*\.\)[^]]*\].*/branch=\1/p' >> $GITHUB_OUTPUT
+
+          # get major version prefix from the release branch ("27.x -> "27.")
+          echo "$TARGET_BRANCH" | sed -n 's/^\([0-9]*\.\).*/target_branch=\1/p' >> $GITHUB_OUTPUT
 
       - name: Check release branch
-        if: github.event.pull_request.base.ref != steps.title_branch.outputs.branch && !(github.event.pull_request.base.ref == 'master' && steps.title_branch.outputs.branch == '')
+        if: steps.title_branch.outputs.target_branch != steps.title_branch.outputs.branch && !(github.event.pull_request.base.ref == 'master' && steps.title_branch.outputs.branch == '')
         run: echo "::error::PR title suggests targetting the ${{ steps.title_branch.outputs.branch }} branch, but is opened against ${{ github.event.pull_request.base.ref }}" && exit 1
```

Comment:
```suggestion
        run: echo "::error::PR is opened against the ${{ github.event.pull_request.base.ref }} branch, but its title suggests otherwise." && exit 1
```

Easiest way out 😅